pkg/controller/common/version/version.go

```diff
@@ -23,6 +23,11 @@ var (
 	SupportedBeatVersions             = MinMaxVersion{Min: From(7, 0, 0), Max: From(8, 99, 99)}
 )
 
+var (
+	// MinDefaultSecurityContextVersion is the minimal version for which the operator will inspect set-default-security-context flag.
+	MinDefaultSecurityContextVersion = MustParse("8.0.0")
```

Comment:
I think I would move that variable to `pkg/controller/elasticsearch/nodespec/podspec.go, very close to the `defaultFsGroup` const. This way we have everything at a single place, and we can write a comment with all the details that may be lost in space otherwise:

```go
// Starting 8.0.0, the Elasticsearch container does not run with the root user anymore. As a result,
// we cannot chown the mounted volumes to the right user (id 1000) in an init container.
// Instead, we can rely on Kubernetes `securityContext.fsGroup` feature: by setting it to 1000
// mounted volumes can correctly be accessed by the default container user.
// On some restricted environments (custom PSPs or Openshift), setting the Pod security context
// is forbidden: the user can either set `--set-default-security-context=false`, or override the
// podTemplate securityContext to an empty value.
var minDefaultSecurityContextVersion = version.MustParse("8.0.0")
const defaultFsGroup = 1000
```